management-proxy/session-manager/routes/system.js

```diff
@@ -58,8 +58,8 @@ module.exports = function (router) {
           if (response.statusCode == HttpStatus.OK && result.namespace) {
             var get_image_id_for_key = result.keys.reduce(function (obj, key) {
               // only compare images belonging to the current channel
-              // this takes care of images like ibuildthecloud/..., too
-              if (key.indexOf(channel) >= 0) {
+              // exclude buildstep as it's currently only available w/ tag :latest.
+              if (key.match(channel + "$") == channel && key.indexOf('buildstep:') === -1) {
```

Comment:
Does this solve the problem?

With ```indexOf``` you start the search for the channel "at the beginning" of the string. Now you are basically start  "at the end" of the key by using the Regex ```WILDCARD$``` resulting in ```":alpha" == ":fooalpha"``` when searching for channel ```alpha```.

I don't know your naming conventions but can you simply split at ```:``` and compare the second part of the array with the channel directly?